koap-core/src/commonTest/kotlin/MessageTest.kt

```diff
@@ -25,11 +25,52 @@ import com.juul.koap.Message.Code.Response.ServiceUnavailable
 import com.juul.koap.Message.Code.Response.Unauthorized
 import com.juul.koap.Message.Code.Response.UnsupportedContentFormat
 import com.juul.koap.Message.Code.Response.Valid
+import com.juul.koap.Message.Option.Accept
+import com.juul.koap.Message.Option.ContentFormat
+import com.juul.koap.Message.Option.ETag
+import com.juul.koap.Message.Option.Echo
+import com.juul.koap.Message.Option.IfMatch
+import com.juul.koap.Message.Option.IfNoneMatch
+import com.juul.koap.Message.Option.Observe
+import com.juul.koap.Message.Option.Observe.Registration.Deregister
+import com.juul.koap.Message.Option.Observe.Registration.Register
+import com.juul.koap.Message.Option.RequestTag
+import com.juul.koap.Message.Option.UriHost
+import com.juul.koap.Message.Option.UriPort
 import kotlin.test.Test
 import kotlin.test.assertEquals
 
 class MessageTest {
 
+    @Test
+    fun optionToStringOverrides() {
+        // content format
+        assertToString(ContentFormat.JSON, "ContentFormat(JSON)")
+        assertToString(ContentFormat.CBOR, "ContentFormat(CBOR)")
+        assertToString(ContentFormat.PlainText, "ContentFormat(PlainText)")
+        assertToString(ContentFormat(20), "ContentFormat(20)")
+        assertToString(Accept(ContentFormat.JSON), "Accept(JSON)")
+
+        // hex
+        assertToString(ETag("123".encodeToByteArray()), "ETag(etag=31 32 33)")
+        assertToString(IfMatch("123".encodeToByteArray()), "IfMatch(etag=31 32 33)")
+        assertToString(IfMatch("".encodeToByteArray()), "IfMatch(etag=)")
+        assertToString(Echo("echo".encodeToByteArray()), "Echo(value=65 63 68 6F)")
```

Comment:
It may be easier to copy/paste if the hex format was w/o spaces.
The default `HexFormat` is lowercase as well, I think that is sensible and we should probably update to align with that.
Perhaps an `0x` prefix would be helpful to identify it as hex while still making copy/paste easy.

```suggestion
        assertToString(Echo("echo".encodeToByteArray()), "Echo(0x6563686f)")
```

Reply:
this just uses `ByteArray.toHexString()` from `Debug.kt`.
in #341, it actually uses `@ExperimentalStdlibApi`, which also implements `toHexString()` but without spaces, which I also think might be better here.

> default HexFormat is lowercase as well

I've come to like uppercase hex more and more for readability. from old c there's no real default, since you choose `%x` or `%X` in printf:s yourself… 😄